app/BUILD.bazel

```diff
@@ -513,10 +514,773 @@ kt_android_library(
+        "src/test/java/DataBinderMapperImpl.java",
+        "src/sharedTest/java/org/oppia/app/utility/OrientationChangeAction.kt"
+    ],
+    test_class = "org.oppia.app.profile.AdminPinActivityTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "PinPasswordActivityTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/profile/PinPasswordActivityTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.profile.PinPasswordActivityTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "ProfileChooserFragmentTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/profile/ProfileChooserFragmentTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.profile.ProfileChooserFragmentTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "ProfilePictureActivityTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/profileprogress/ProfilePictureActivityTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.profileprogress.ProfilePictureActivityTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "ProfileProgressFragmentTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/profileprogress/ProfileProgressFragmentTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.profileprogress.ProfileProgressFragmentTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "BindableAdapterTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/recyclerview/BindableAdapterTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.recyclerview.BindableAdapterTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "ProfileEditActivityTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/settings/profile/ProfileEditActivityTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.settings.profile.ProfileEditActivityTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "ProfileListActivityTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/settings/profile/ProfileListActivityTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.settings.profile.ProfileListActivityTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "ProfileRenameActivityTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/settings/profile/ProfileRenameActivityTest",
+        "src/test/java/DataBinderMapperImpl.java"
+    ],
+    test_class = "org.oppia.app.settings.profile.ProfileRenameActivityTest",
+    deps = TEST_DEPS,
+)
+
+app_test(
+    name = "ProfileResetPinActivityTest",
+    srcs = [
+        "update_src/sharedTest/java/org/oppia/app/settings/profile/ProfileResetPinActivityTest",
```

Comment:
I just figured that a target name shouldn't have the `.kt` notation - but in this particular case since it is being used as a source I am okay with changing it! Should I go ahead and do that?

Reply:
Please see my newer comment & discuss there. I think if we can hide the implementation details of the genrule it should be fine.